Use logstash redis defaults.

jmcarp opened the pull request:
[Resolves #24]

Most of the diffs here are changing tabs to spaces. The only meaningful change is dropping the default redis congestion properties, then only adding them to the redis options if defined.

jmcarp committed:
Use logstash redis defaults.

[Resolves #24]

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cloudfoundry-community/logsearch-boshrelease?pullRequest=25) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cloudfoundry-community/logsearch-boshrelease?pullRequest=25) before we can accept your contribution.<br/>

rogeruiz commented on jobs/ingestor_syslog/templates/config/logstash.conf.erb:
```diff
@@ -67,35 +67,39 @@ output {
+          if_p('redis.congestion_threshold') do |threshold|
+            output['options']['congestion_threshold'] = threshold
+          end
+        %>
```
72 to 87 are the changes introduced in this PR. Most of the rest is whitespace.

Infra-Red reviewed (changes requested):
Hi @jmcarp, everything looks fine, but could you please update redis-input-plugin for parser logstash instances to also use the ELK default settings.

jmcarp committed:
Drop non-default options and allow arbitrary option overrides.

jmcarp committed:
Fix unit tests.

jmcarp commented:
The only non-default option I saw for redis input was `threads`, which I moved into the generic `options` hash, which I updated to be merged into the individually defined options. Let me know if I missed anything else!

Infra-Red commented:
LGTM
